utils/material_plots_2D.py

```diff
@@ -1,80 +1,167 @@
+"""
+This script must be called with python: 'python material_scan_2D.py --{argument} {value}'.
+The output files are saved in data/{outputDir}/name.suffix.
+If no outputDir is specified, it will be data/plots/name.suffix.
+"""
+
 from __future__ import print_function
+
 import argparse
 import math
+import sys
+from os import fspath
+from os.path import expandvars
+from pathlib import Path
 
-import sys,os
-sys.path.append(os.path.expandvars("$FCCSW") + "/Examples/scripts")
-from plotstyle import FCCStyle
 import ROOT
 
+sys.path.append(expandvars("$FCCSW") + "/Examples/scripts")
+from plotstyle import FCCStyle
+
 
 def main():
-    parser = argparse.ArgumentParser(description='Material Plotter')
-    parser.add_argument('--fname', "-f", dest='fname', type=str, help="name of file to read")
-    parser.add_argument('--angleMin', dest='angleMin', default=6, type=float, help="minimum eta/theta/cosTheta")
-    parser.add_argument('--angleMax', dest='angleMax', default=6, type=float, help="maximum eta/theta/cosTheta")
-    parser.add_argument('--angleDef', dest='angleDef', default="eta", type=str, help="angle definition to use: eta, theta or cosTheta, default: eta")
-    parser.add_argument('--angleBinning', "-b", dest='angleBinning', default=0.05, type=float, help="eta/theta/cosTheta bin width")
-    parser.add_argument('--nPhiBins', dest='nPhiBins', default=100, type=int, help="number of bins in phi")
-    parser.add_argument('--x0max', "-x", dest='x0max', default=0.0, type=float, help="Max of x0")                                                                                                                                                
+    parser = argparse.ArgumentParser(description="Material Plotter")
```

Comment:
Can we please not mix style changes and feature changes?

Reply:
That was not my intention. I have activated auto-formatting (black + isort) in code and as soon as I open a script it is also formatted. In general, I think autoformatting makes a lot of sense in a project like this...

Reply:
The issue in this case is not the re-formatting itself, but rather that the formatting changes and feature changes are mixed in one commit. In case there is an auto-formatter, the way to go would be: auto-format only first in a separate commit and then make a second commit with the feature changes.

That makes it easier to review the feature changes only, because we have a diff with only that. Additionally, git can be configured to ignore format only commits in `git blame`, which again makes it easier to figure out what actually changed in the future.

Reply:
> That was not my intention. I have activated auto-formatting (`black` + `isort`) in code and as soon as I open a script it is also formatted. In general, I think autoformatting makes a lot of sense in a project like this...

To put it bluntly and exaggerating :): No, it makes no sense like this, everyone using a different auto linter leads to way too much noise. When in Rome do as the Romans. When there is no linter config, don't lint by default.

So in this case the issue is the re-formatting :)

Yes auto-formatting is nice, but sadly there is a cost for "legacy" code.

Reply:
I see. I think I managed to add a commit that isolates the style changes :)